MSAL/src/public/MSALPublicClientApplication.h

```diff
@@ -34,50 +34,156 @@
+ */
 - (id)initWithClientId:(NSString *)clientId
                  error:(NSError * __autoreleasing *)error;
+
+/*!
+    Initialize a MSALPublicClientApplication with a given clientID and authority
+ 
+    @param  clientId    The clientID of your application, you should get this from the app portal.
+    @param  authority   The authority to use for the application
+    @param  error       The error that occurred creating the application object, if any, if you're
+                        not interested in the specific error pass in nil.
+ */
 - (id)initWithClientId:(NSString *)clientId
              authority:(NSString *)authority
                  error:(NSError * __autoreleasing *)error;
 
+/*!
+    Returna an array of users visible to this application
+ 
+    @param  error   The error that occured trying to retrieve users, if any, if you're
+                    not interested in the specific error pass in nil.
+ */
 - (NSArray <MSALUser *> *)users:(NSError * __autoreleasing *)error;
+
+/*!
+    Returns a specific user for the identifier given (received from a user object returned
+    in a previous acquireToken call)
+ 
+    @param  error   The error that occured trying to the user, if any, if you're
+                    not interested in the specific error pass in nil.
+ */
 - (MSALUser *)userForIdentifier:(NSString *)identifier
                           error:(NSError * __autoreleasing *)error;
 
 #pragma SafariViewController Support
 
+/*!
+    Returns whether or not the URL is a response to a MSAL web authentication
+    session. Call this before passing the URL into -handleMSALResponse:
+ */
 + (BOOL)isMSALResponse:(NSURL *)response;
+
+/*!
+    Pass a URL response from your application delegate's openURL handler into
+    MSAL for web authnetication sessions.
+ */
 + (void)handleMSALResponse:(NSURL *)response;
+
+/*!
+    Cancel a currently running interactive web authentication session.
+ */
 + (void)cancelCurrentWebAuthSession;
 
 #pragma mark -
 #pragma mark acquireToken
 
+/*!
+    Acquire a token for a new user using interactive authentication
+ 
+    @param  scopes          Scopes to request from the server, the scopes that come back
+                            can differ from the ones in the original call
+    @param  completionBlock The completion block that will be called when the authentication
+                            flow completes, or encounters an error.
+ */
 - (void)acquireTokenForScopes:(NSArray<NSString *> *)scopes
               completionBlock:(MSALCompletionBlock)completionBlock;
 
 #pragma mark -
 #pragma mark acquireToken using Login Hint
 
+
+/*!
+    Acquire a token for a new user using interactive authentication
+ 
+    @param  scopes          Scopes to request from the server, the scopes that come back
+                            can differ from the ones in the original call
+    @param  loginHint       A loginHint (usually an email) to pass to the service at the
```

Comment:
It's never needed. The only case where it'd be used is if you previously collected the user's email in some other piece of UI.